install.sh

```diff
@@ -0,0 +1,198 @@
+    echo "  -h: Show this help message and exit"
+    echo "  -d: Install only dependencies"
+    echo "  -p: Install only python dependencies (including playwright)"
+}
+
+
+
+while getopts ":hydp" option; do
+    case $option in
+        h)
+            Help
+            exit 0;;
+        y)
+            ASSUME_YES=1;;
+        d)
+            DEPS_ONLY=1;;
+        p)
+            PYTHON_ONLY=1;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            Help
+            exit 1;;
+        :)
+            echo "Option -$OPTARG requires an argument." >&2
+            Help
+            exit 1;;
+    esac
+done
+
+
+
+function install_macos(){
+    if [ -z "$(command -v brew)" ]; then
+        NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+        if [[ uname -m == "x86_64" ]]; then
+            echo "export PATH=/usr/local/bin:$PATH" >> ~/.bash_profile && source ~/.bash_profile
+        else
+            echo "export PATH=/opt/homebrew/bin:$PATH" >> ~/.bash_profile && source ~/.bash_profile
+        fi
+    else
+        echo "Homebrew is already installed"
+    fi
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        brew install python@3.10
+    else
+        echo "Python is already installed"
+    fi    
+}
+
+function install_arch(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo pacman -S python-pip
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo pacman -S git
+    else
+        echo "Git is already installed"
+    fi
+}
+
+function install_deb(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo apt-get install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo apt-get install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo apt-get install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function install_fedora(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo dnf install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo dnf install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo dnf install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function install_centos(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo yum install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo yum install python3-pip
```

Comment:
What about using ensurepip instead of installing it separately 
```suggestion
        python3 -m ensurepip --upgrade
```